### PR TITLE
Class Variables with Default value

### PR DESCRIPTION
**Pull request by MarcusDenker**
InitializedClassVariable and LazyClassVariable, with tests

#name => LazyClassVariable default: 5.

can now be used to create a class var with default value

### CODE REVIEW

**Review by dionisiydk — COMMENTED** [review]
Hi Marcus.

I think it would be nice to hide those classes from the user:
```Smalltalk
var => ClassVariable default: 5
```
It will produce InitializedClassVariable.
And for lazy:
```Smalltalk
var => ClassVariable lazy default: 5
```

File: src/Slot-Examples/AbstractInitializedClassVariable.class.st [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		'default',
+		'isBlock'
+	],
+	#category : #'Slot-Examples-ClassVariables'
+}
+
+{ #category : #accessing }
+AbstractInitializedClassVariable >> default: anObject [
+	default := anObject.
+	isBlock := false
```
Comment (dionisiydk) [inline src/Slot-Examples/AbstractInitializedClassVariable.class.st:21]: isBlock variable looks redundant.
If there is anyway isBlock ifTrue logic then just ask "default isBlock"

Comment (MarcusDenker) [reply src/Slot-Examples/AbstractInitializedClassVariable.class.st:21]: I for now did it in a way that you can specify a block as the default, but not evaluate it.
AbstractInitializedClassVariable default: []
vs
AbstractInitializedClassVariable defaultBlock: []

in the second case the block is evaluated, in the first not

I was trying to do it by always wrapping in a block, but that is not nice either.

**Comment by MarcusDenker** [comment]
The problem with

var => ClassVariable default: 5

is that there will be other ClassVariables, too. And I wanted to follow the style of Slots for now...

**Comment by MarcusDenker** [comment]
(I want in the end a unified way:

#myVar => Observable + Lazy default: 5

should work for both slots and class vars... )

File: src/Slot-Examples/AbstractInitializedClassVariable.class.st (hunk 1 above)
Comment (MarcusDenker) [reply src/Slot-Examples/AbstractInitializedClassVariable.class.st:21]: But I think it is possible to simplify: just have default:, and if people want a block they have to do [[]] 

I will try that...

Comment (jecisc) [reply src/Slot-Examples/AbstractInitializedClassVariable.class.st:21]: Can’t default be 

default: anObject
    self defaultBlock: [ anObject ]

?

Or the other way around. I am on my phone and it’s not easy to read the code.

**Comment by MarcusDenker** [comment]
I made a PR with a version like Denis suggested: https://github.com/pharo-project/pharo/pull/4420

**Pull request merged**